src/interrupt.c

```diff
@@ -314,10 +318,29 @@ void __MI_handler(void)
  */
 void __TI_handler(void)
 {
-	/* timer int cleared in int handler */
+	/* NOTE: the timer interrupt is already acknowledged in inthandler.S */
     __call_callback(TI_callback);
 }
 
+/**
+ * @brief Handle a CART interrupt
+ */
+void __CART_handler(void)
+{
+    /* Call the registered callbacks */
+    __call_callback(CART_callback);
+
+     /* CART interrupts must be acknowledged by handlers. If the handler fails
+       to do so, the console freezes because the interrupt will retrigger
+       continuously. Since a freeze is always bad for debugging, try to 
+       detect it, and show a proper assertion screen. */
+    if (!(C0_CAUSE() & C0_INTERRUPT_CART))
```

Comment:
Should we also keep these behind `#ifndef NDEBUG` to reduce a little bit of code at final builds?

Reply:
Sure, done thanks.